environment/src/main/java/com/sequenceiq/environment/environment/service/LoadBalancerPollerService.java

```diff
@@ -0,0 +1,165 @@
+            .stopIfException(pollingConfig.getStopPollingIfExceptionOccured())
+            .waitPeriodly(pollingConfig.getSleepTime(), pollingConfig.getSleepTimeUnit())
+            .run(() -> periodicCheckForCompletion(flowIdentifiers));
+    }
+
+    private AttemptResult<List<FlowIdentifier>> periodicCheckForCompletion(List<FlowIdentifier> flowIdentifiers) {
+        try {
+            boolean anyFlowsActive = false;
+            for (FlowIdentifier flowIdentifier: flowIdentifiers) {
+                Boolean hasActiveFlow = ThreadBasedUserCrnProvider.doAsInternalActor(() ->
+                    flowEndpoint.hasFlowRunningByFlowId(flowIdentifier.getPollableId()).getHasActiveFlow());
+                if (hasActiveFlow) {
+                    LOGGER.debug("Flow {} is still running", flowIdentifier.getPollableId());
+                } else {
+                    LOGGER.debug("Flow {} is complete", flowIdentifier.getPollableId());
+                }
+                anyFlowsActive = anyFlowsActive || hasActiveFlow;
+            }
+            if (anyFlowsActive) {
+                return AttemptResults.justContinue();
+            } else {
+                List<FlowIdentifier> failedFlows = flowIdentifiers.stream()
+                    .filter(flowId -> hasFlowFailed(ThreadBasedUserCrnProvider.doAsInternalActor(() ->
+                        flowEndpoint.getFlowLogsByFlowId(flowId.getPollableId()))))
+                    .collect(Collectors.toList());
+                return AttemptResults.finishWith(failedFlows);
+            }
+        } catch (Exception e) {
+            LOGGER.warn("Failure checking status of flows {}, error is: {}", flowIdentifiers,
+                e.getMessage());
+            return AttemptResults.breakFor(e);
+        }
+    }
+
+    private PollingConfig getPollingConfig() {
+        return PollingConfig.builder()
+            .withStopPollingIfExceptionOccured(true)
+            .withSleepTime(sleepTime)
+            .withSleepTimeUnit(TimeUnit.SECONDS)
+            .withTimeout(maxTime)
+            .withTimeoutTimeUnit(TimeUnit.SECONDS)
+            .build();
+    }
+
+    private Set<String> getDataLakeAndDataHubNames(String environmentCrn, String environmentName) {
+        return Stream.of(getAttachedDatahubClusters(environmentCrn), getAttachedDatalakeClusters(environmentName))
+            .flatMap(Collection::stream)
+            .collect(Collectors.toSet());
+    }
+
+    private Set<String> getAttachedDatahubClusters(String environmentCrn) {
+        LOGGER.debug("Getting Datahub clusters for environment: '{}'", environmentCrn);
+        Collection<StackViewV4Response> responses = datahubService.list(environmentCrn).getResponses();
+        return responses.stream().map(StackViewV4Response::getName).collect(Collectors.toSet());
+    }
+
+    private Set<String> getAttachedDatalakeClusters(String environmentName) {
+        LOGGER.debug("Getting SDX clusters for environment: '{}'", environmentName);
```

Comment:
In the logs there are Datahubs/Data Hubs/Datalakes/Data Lakes/SDX can we add this terms consistent ?